test/built-ins/Array/prototype/groupBy/callback-arg.js

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2021 Ecma International.  All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.groupBy
+description: Array.prototype.groupBy populates Map with correct keys and values
+info: |
+  22.1.3.14 Array.prototype.groupBy ( callbackfn [ , thisArg ] )
+
+  ...
+
+  6. Repeat, while k < len
+
+      a. Let Pk be ! ToString(𝔽(k)).
+      b. Let kValue be ? Get(O, Pk).
+      c. Let propertyKey be ? ToPropertyKey(? Call(callbackfn, thisArg, « kValue, 𝔽(k), O »)).
+      d. Perform ! AddValueToKeyedGroup(groups, propertyKey, kValue).
+      e. Set k to k + 1.
+  ...
+
+---*/
+
+
+const arr = [-0,0,1,2,3];
+
+const sentinel = {};
+
+const map = arr.groupBy.call(sentinel, (n,i,testArr) => {
```

Comment:
IIUC, `LengthOfArrayLike` of the `sentinel` is 0, the callback won't be called in this case.